src/objective-c/GRPCClient/GRPCCall.m

```diff
@@ -193,14 +191,17 @@ - (void)startNextRead {
   }
   __weak GRPCCall *weakSelf = self;
   __weak GRPCDelegateWrapper *weakWriteable = _responseWriteable;
-
+  
   dispatch_async(_callQueue, ^{
-    [weakSelf startReadWithHandler:^(grpc_event *event) {
-      if (!event->data.read) {
-        // No more responses from the server.
+    [weakSelf startReadWithHandler:^(NSDictionary *event) {
+      NSData *data = event[[NSNumber numberWithInt:GRPC_OP_RECV_MESSAGE]];
+      if (data == [NSNull null]) {
```

Comment:
We won't be needing that dictionary anymore anyway, but for completeness: If we just don't put the entry in the dictionary, `dict[key]` would return `nil`. So `[NSNull null]` is only necessary when you want to treat the presence or absence of the entry differently.